tests/QueryBuilderTest.php

```diff
@@ -449,25 +449,34 @@ public function testDistinct()
 
     public function testCustomId()
     {
+        $tags = [['id' => 'sharp', 'name' => 'Sharp']];
         DB::table('items')->insert([
-            ['id' => 'knife', 'type' => 'sharp', 'amount' => 34],
-            ['id' => 'fork', 'type' => 'sharp', 'amount' => 20],
+            ['id' => 'knife', 'type' => 'sharp', 'amount' => 34, 'tags' => $tags],
+            ['id' => 'fork', 'type' => 'sharp', 'amount' => 20, 'tags' => $tags],
             ['id' => 'spoon', 'type' => 'round', 'amount' => 3],
         ]);
 
         $item = DB::table('items')->find('knife');
         $this->assertEquals('knife', $item->id);
+        $this->assertObjectNotHasProperty('_id', $item);
+        $this->assertEquals('sharp', $item->tags[0]['id']);
+        $this->assertArrayNotHasKey('_id', $item->tags[0]);
 
         $item = DB::table('items')->where('id', 'fork')->first();
         $this->assertEquals('fork', $item->id);
 
+        // tags.id is translated into tags._id in query
```

Comment:
There are plenty of assertions in this test for `_id` _not_ existing on the model object(s), but what actually asserts that `_id` is stored in the database? Presumably, the query below would work even if `id` was never translated to `_id` on the way to the server.

If it's not the intention to test that here, feel free to resolve. This comment made me think otherwise.

Reply:
I added a query on `_id` field in this test.